src/constraint_functions.jl

```diff
@@ -59,22 +66,31 @@ If you create new capacity variables, it is beneficial to include as well a meth
 function and the corresponding node types.
 """
 function constraints_capacity_installed(m, n::Node, 𝒯::TimeStructure, modeltype::EnergyModel)
-
-    cap = capacity(n)
-    @constraint(m, [t ∈ 𝒯],
-        m[:cap_inst][n, t] == cap[t]
-    )
+    # Fix the installed capacity to the upper bound
+   for t ∈ 𝒯
+        fix(m[:cap_inst][n, t], capacity(n, t); force=true)
+   end
 end
 function constraints_capacity_installed(m, n::Storage, 𝒯::TimeStructure, modeltype::EnergyModel)
+    # Extract the required fields from the composite type
+    par_charge = charge(n)
+    par_level = level(n)
+    par_discharge = discharge(n)
 
-    cap = capacity(n)
-    @constraint(m, [t ∈ 𝒯],
-        m[:stor_cap_inst][n, t] == cap.level[t]
-    )
-
-    @constraint(m, [t ∈ 𝒯],
-        m[:stor_rate_inst][n, t] == cap.rate[t]
-    )
+    # Fix the installed capacity to the upper bound
+    for t ∈ 𝒯
+        fix(m[:stor_level_inst][n, t], capacity(par_level, t); force=true)
+    end
+    if isa(par_charge, UnionCapacity)
+        for t ∈ 𝒯
+            fix(m[:stor_charge_inst][n, t], capacity(par_charge, t); force=true)
+        end
+    end
+    if isa(par_discharge, UnionCapacity)
```

Comment:
same here, would be nice with a more readable utility function name or dispatch to subfunctions?

Reply:
See above.